addons/website_sale/controllers/main.py

```diff
@@ -240,9 +247,23 @@ def shop(self, page=0, category=None, search='', ppg=False, **post):
                 parent_category_ids.append(current_category.parent_id.id)
                 current_category = current_category.parent_id
 
-        product_count = Product.search_count(domain)
-        pager = request.website.pager(url=url, total=product_count, page=page, step=ppg, scope=7, url_args=post)
-        products = Product.search(domain, limit=ppg, offset=pager['offset'], order=self._get_search_order(post))
+        # product search in minimum price and maximum price all of product
+        if max_price or min_price:
+            products = Product.search(domain, limit=False, order=self._get_search_order(post))
+            if max_price and min_price:
+                products = products.filtered(lambda p: p.website_price >= float(min_price) and p.website_price <= float(max_price))
+            elif max_price:
+                products = products.filtered(lambda p: p.website_price <= float(max_price))
+            elif min_price:
+                products = products.filtered(lambda p: p.website_price >= float(min_price))
```

Comment:
Wouldn't it be smarter to add these filters to the domain? Would also avoid having to retrieve half the database and senselessly screw with pagination.